infrastructure/code/code_test.go

```diff
@@ -892,3 +896,159 @@ func Test_SastApiCall(t *testing.T) {
 		assert.Equal(t, err.Error(), "SAST is not enabled")
 	})
 }
+
+func TestScanner_getFilesToBeScanned(t *testing.T) {
+	config.CurrentConfig().SetSnykCodeEnabled(true)
+	_, scanner := setupTestScanner(t)
+	tempDir := t.TempDir()
+	scanner.changedPaths = make(map[string]map[string]bool)
+	scanner.changedPaths[tempDir] = make(map[string]bool)
+
+	t.Run("should add all files from changedPaths map and delete them from changedPaths", func(t *testing.T) {
+		changedFile := "file1.java"
+		scanner.changedPaths[tempDir][changedFile] = true
+		scanner.changedPaths[tempDir]["file2.java"] = true
+
+		files := scanner.getFilesToBeScanned(tempDir)
+
+		require.Contains(t, files, changedFile)
+		require.Contains(t, files, "file2.java")
+		require.Len(t, scanner.changedPaths[tempDir], 0)
+	})
+
+	t.Run("should add all files that have dataflow items of a changed file", func(t *testing.T) {
+		changedFile := "main.ts"
+		fromChangeAffectedFile := "juice-shop/routes/vulnCodeSnippet.ts"
+
+		// we need to add a fake issue to the cache, let's keep it to the minimum needed
+		// we are reusing the test data from the code_html test
```

Comment:
I can see us changing some of the logic in that code_html file or test and then we break this. Would it not be better and clearer to just hardcode it here? It'll also make it more obvious why this test does what it does

Reply:
Yeah, I was lazy ;)